usecases/telemetry/telemetry.go

```diff
@@ -0,0 +1,229 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package telemetry
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"runtime"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/verbosity"
+	"github.com/weaviate/weaviate/usecases/config"
+)
+
+const (
+	defaultConsumerURL  = "https://us-central1-semi-production.cloudfunctions.net/weaviate-telemetry"
```

Comment:
what do you think if we make it configurable instead so that we can address the region and the env ? 
something like 

```go
fmt.Sprintf("https://%s-%s/weaviate-telemetry", os.Getenv("REGION"), os.Getenv("ENV_URL"))
```

Reply:
This URL is only temporary, will soon be placed by a subdomain of weaviate.io

Reply:
out of curiosity is that a public available URL ?   I am asking because we could be spammed by sniffers our in the wild :D